can-auth/vatican.c

```diff
@@ -255,3 +358,62 @@ int VULCAN_FUNC vulcan_recv(ican_t *ican, uint16_t *id, uint8_t *buf, int block)
     vatican_commit_nonce_increment();
     return rv;
 }
+
+/*******************************************************************/
+/* IAT nonce synchronisation */
+/*******************************************************************/
+
+#if VATITACAN
+void VULCAN_FUNC iat_send_callback(void)
+{
+    timer_disable();
+    sleeping = 0x0;
+}
+#endif
+
+#if VATITACAN
+long VULCAN_FUNC encode_iat(uint32_t nonce)
+{
+    uint32_t masked;
+
+    // Retrieve last 2 nonce bits
+    masked = nonce & nonce_mask;
+
+    return (VATITACAN_OFFSET + masked * VATITACAN_DELTA);    
+}
+#endif
+
+#if VATITACAN
+void VULCAN_FUNC iat_recv_callback(void)
```

Comment:
annotating this as VULCAN_FUNC will not work when it's an actual enclave -- interrupts cannot call private enclave functions